bokehjs/src/coffee/model.coffee

```diff
@@ -17,6 +21,42 @@ export class Model extends HasProps
       for cb in callbacks
         @listenTo(@, evt, () -> cb.execute(@))
 
+    @listenTo(@, 'change:js_event_callbacks', () -> @_update_event_callbacks)
+
+
+  _process_event : (event) ->
+    # Given an applicable event, execute the associated callback
+    if not event.constructor.event_name
+       logger.warn('Event is still passed as a string: ' + event)
+       return
+    if not event.constructor.applicable_models.some((m) => m == @.type)
```

Comment:
Events aren't Models. This is something I discussed with @bryevdv and we didn't find any strong reason to make them into models. In addition, ``applicable_models`` can be a list so one event may have multiple applicable models and I'm not entirely sure the appropriate specification would always follow the structure of the inheritance hierarchy.